telematic_system/scripts/log_analysis/parse_processing_service_logs.py

```diff
@@ -0,0 +1,82 @@
+import re
+import sys
+
+from enum import Enum
+import datetime
+import pytz
+import numpy as np
+
+class ProcessingStatus(Enum):
+    IN_PROGRESS= "IN_PROGRESS"
+    COMPLETED= "COMPLETED"
+    ERROR= "ERROR"
+
+
+def to_utc_datetime(self, timestamp):
+        return datetime.datetime.fromtimestamp(timestamp).astimezone(pytz.utc).replace(tzinfo=None)
+
+def parse_start_end_processing_line(line, search_str):
+    # Regex match to extract timestamp of the form: "log":"2024-07-23 15:43:45,821
+    timestamp_match = re.search(r'"log":"(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d+)', line)
+    if timestamp_match:
+        timestamp_str = timestamp_match.group(1)
+    else:
+        print("Could not find processing starting time for rosbag")
+
+    log_timestamp = datetime.datetime.strptime(timestamp_str, '%Y-%m-%d %H:%M:%S,%f')
+
+
+    # Extract the rosbag name
+    rosbag_match = re.search(fr'{search_str} (\S+\.mcap)', line)
+    rosbag_name = rosbag_match.group(1) if rosbag_match else None
+
+    return log_timestamp, rosbag_name
+
+
+def parse_processing_service_logs(logfile):
+    processing_started_msg_str = "Processing rosbag:"
+    processing_completed_msg_str = "Completed rosbag processing for:"
+    processing_status_str = "Updated mysql entry for"
+
+    processing_times = []
+
+    with open(logfile, 'r') as file:
+        for line in file:
+            if processing_started_msg_str in line:
```

Comment:
Add a sample start message from the log?
The same comments for the end message and mysql entry?